File: 2022/tests/day20.test.ts
import { readFileSync } from 'fs';
import { part1, part2 } from '../src/day20';

const dayNumber = __filename.match(/day(\d+)\.test\.ts/)?.[1];
const inputFile: string = `input/day${dayNumber}.txt`;

const example: string = `
1
2
-3
3
-2
0
4
`.trim();
const input: string = readFileSync(inputFile, 'utf8');

describe('part 1', () => {
  test('example', () => {
    expect(part1(example)).toBe(3);
  });
  test(`solution`, () => {
    expect(part1(input)).toBe(4224);
  });
});

describe('part 2', () => {
  test('example', () => {
    expect(part2(example)).toBe(1623178306);
  });
  test(`solution`, () => {
    expect(part2(input)).toBe(861907680486);
  });
});

File: 2024/AdventOfCode/Tests/Day20Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day20Test {
    private const string Example = "Inputs/20-Example.txt";
    private const string Input = "Inputs/20.txt";
    private static Day20 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day20(Example);
            const string expected = "0";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day20(Input);
            const string expected = "1286";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day20(Example);
            const string expected = "0";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day20(Input);
            const string expected = "989316";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}